### ISSUE TITLE
Config file precedence, staging configs, etc.

### ISSUE THREAD

**Issue by jcswart** [regolith-linux/regolith-desktop#315]
First off: amazing work on regolith it is a really wonderful project and I can't wait to use it more. That said it has a few issues, for me personally, that I need help resolving.

1. The rofi `-width` option seems to have no effect. It isn't listed at https://regolith-linux.org/docs/reference/xresources/ so I assume I must handle it elsewhere?
    * For clarity if run `rofi -show drun -width $ANY_VALUE` from a terminal there is never any effect for any value of $ANY_VALUE (ex: 20, 50, 1920) etc.
    * Also this seems to be something of a global. The notification window is the same, which makes me think that something in regolith or elsewhere is "trapping/containing" this.
2. The window highlight, as often mentioned, is not good for me personally. Yet I am having zero luck changing it following the various other topics mentioned.
    * I have seen https://github.com/regolith-linux/regolith-desktop/issues/210 and https://github.com/regolith-linux/regolith-desktop/issues/42 and not had luck 

Which brings me to my sort of overall question of the issue title.

__What is the actual precedence of config values, and what is the way of figuring out how to actually modify values in an idiomatic way?__

It seems the Xresources file https://regolith-linux.org/docs/reference/xresources/ is the idiom as of 1.3, but that only works if the value is supported. For example altering the global "meta/mod" key is not supported. That was the first thing I had to change and so now I have a staged i3 config. Which according to the list at https://regolith-linux.org/docs/reference/configurations/ will overwrite other values?

I'm a little confused. Is the Xresource file last, or overwritten by staged configs?

Really I just want `rofi` to respect the width command because I am on HiDPI so I have to increase the font to make it usable in the first place, but that then takes up the whole tiny window.

And second I just want a very obvious border around the active pane (terminal, browser, etc), similar to the i3 default. 

Any pointers?

**Comment by kgilmer** [comment]
Hi @jcswart :smile: 

You can see the load order of Xresource and i3 files here: https://regolith-linux.org/docs/reference/configurations/#load-sequence.

Generally speaking, if you just want to override existing values, then `~/.config/regolith/Xresources` is your best bet, because you only need to specify your delta.  However if you wish to create your own setup, or modify some things that cannot be adjusted via Xresources (such as the keybinding literals) then you need to stage your own i3 config file.

## Rofi

An easy way of seeing what Xresources are available and what they are set to is the `xrdb` command.  Here we can see every setting involving Rofi: `xrdb -query | grep rofi`:

```
$ xrdb -query  | grep rofi
rofi.drun-display-format:       {name}
rofi.drun-icon-theme:   Moka
rofi.font:      JetBrains Mono Medium 16
rofi.modi:      window,drun,run
rofi.show-icons:        true
rofi.theme:     /etc/regolith/styles/ayu-mirage-regolith/rofi.rasi
rofi.window-format:     \ {n} {t}
```

The item I think you're after is the theme.  I believe Rofi is ignoring your command line setting because the theme is set and Rofi is loading that.  So, to make your change, you could copy `/etc/regolith/styles/ayu-mirage-regolith/rofi.rasi` to `~/.config/regolith/rofi/rofi.rasi`, and then add this line to `~/.config/regolith/Xresources`:

```
rofi.theme: /home/<username>/.config/regolith/rofi/rofi.rasi
```

Then make any changes you need to there.  Once done, `regolith-look refresh` should cause Rofi to update.

## Super key

I successfully tested swapping Alt and Win for super with the following, in the `~/.config/regolith/Xresources` file:

```
i3-wm.mod: Mod1
i3-wm.alt: Mod4
```

And then again `regolith-look refresh`

## Window borders

I've just tested the following change to increase the width of borders in the `~/.config/regolith/Xresources` file:

```
i3-wm.window.border.size: 10
```
And then again `regolith-look refresh`, and open a new window to see the change. 

## Other

Despite all of my examples being tested against unstable, I would also recommend not running from `ppa:regolith-linux/unstable` unless you're willing to deal with some breakage.  For example, ayu-mirage-regolith was renamed yesterday.  But, it's always nice to get bug reports from people running `unstable`, so it's your call.  

Please let me know if anything is unclear, and feel free to open an issue for documentation if you feel something could be made more clear, by clicking the "Open a Documentation issue" on the right side of every page.

**Comment by jcswart** [comment]
Thank you so much!

I figured out one aspect of my problem: I set gnome-terminal as my default and the latest version does not appear to play very nicely with Regolith, w.r.t the border. I had been doing the border testing by having several terminal windows open and so I wasn't ever able to detect a change. A window border of 100px was very noticeable on Firefox however.

I need to figure out what causes the "highlight" effect. My suspicion is that the gnome side of regolith is controlling that perhaps... not sure. Essentially when I select something the chrome is highlighted, that might be the best I can get with the gnome-terminal. I'll investigate other options maybe `st` will be fine since I don't have to recompile it each time with regolith for font sizes.

It was not made obvious to me that ANY config value could be overwritten since I didn't see the mod keys on the big list of the Xresources page. I'm rushing to replace a dead system so I can work so I'm not being the most thorough.

Anyway I'm armed with what I need and I put some 💰 in the open collective. Thank you so much for your work on this project. Spacemacs + Regolith mean that I can make relatively low effort linux environments on demand. Super awesome!
